system-tests/example-styx-plugin-dependencies/src/main/java/depend/ExampleDependency.java

```diff
@@ -0,0 +1,26 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package depend;
+
+/**
+ * A project can attempt to reference this to prove the dependency relationship is in effect.
+ */
+public final class ExampleDependency {
+    public static final String EXAMPLE_DEPENDENCY_PROPERTY = "Example-Dependency-Available";
```

Comment:
final Strings are not loaded dynamically but rather compiled inside the dependent byte code.  That is, this line : 
LOGGER.info(EXAMPLE_DEPENDENCY_PROPERTY);  will not try to load the dependent JAR file and obtain the  EXAMPLE_DEPENDENCY_PROPERTY at all:  ff you decompile the bytecode for  TestPluginModule.java  you will see it is just doing "logger.info("Example-Dependency-Available")